### ISSUE TITLE
mktemp provided by easy-rsa for windows is broken

### ISSUE THREAD

**Issue by luizluca** [OpenVPN/easy-rsa#243]
Hello,

I'm testing easy-rsa on windows and mktemp ignores the mask:

```
# a=$(mktemp test.XXXXXX)
path = test.XXXXXX
lpPathBuffer = C:\Users\luizluca\AppData\Local\Temp\
szTempName = C:\Users\luizluca\AppData\Local\Temp\tmpEACF.tmp
path = C:\Users\luizluca\AppData\Local\Temp\tmpEACF.tmp
fd = 4

EasyRSA Shell
# echo $a
test.XXXXXX
```

Even if considered the wrong name, it does not even create the file.

```
# mktemp a.XXXXX
path = a.XXXXX
lpPathBuffer = C:\Users\luizluca\AppData\Local\Temp\
szTempName = C:\Users\luizluca\AppData\Local\Temp\tmp5820.tmp
path = C:\Users\luizluca\AppData\Local\Temp\tmp5820.tmp
fd = 4
a.XXXXX

EasyRSA Shell
# ls a.*
ls: a.*: No such file or directory
```

Only directory seems to actually work:

```
# a=$(mktemp -d key_out_tmp.XXXXXX)

EasyRSA Shell
# echo $a
key_out_tmp.a02876
```

I tested on Win7

**Comment by TinCanTech** [comment]
Looks like `mktemp` on Windows is a failure .. eg. collision:
```
EasyRSA Shell
# mktemp -V
mktemp version 1.6

EasyRSA Shell
# foo="$(mktemp test.XXXXXXXXXX)"
path = test.XXXXXXXXXX
lpPathBuffer = C:\Users\root\AppData\Local\Temp\
szTempName = C:\Users\root\AppData\Local\Temp\tmp2E6F.tmp
path = C:\Users\root\AppData\Local\Temp\tmp2E6F.tmp
fd = 3

EasyRSA Shell
# echo "foo" > "$foo"

EasyRSA Shell
# cat "$foo"
foo

EasyRSA Shell
# bar="$(mktemp test.XXXXXXXXXX)"
path = test.XXXXXXXXXX
lpPathBuffer = C:\Users\root\AppData\Local\Temp\
szTempName = C:\Users\root\AppData\Local\Temp\tmpBA0.tmp
path = C:\Users\root\AppData\Local\Temp\tmpBA0.tmp
fd = 3

EasyRSA Shell
# echo "bar" >> "$bar"

EasyRSA Shell
# cat "$bar"
foo
bar

EasyRSA Shell
# cat "$foo"
foo
bar

EasyRSA Shell
#
```
Which means the template mask is of no consequence.

**Comment by luizluca** [comment]
BTW, #240 uses directories for temp files. As masks in directory path does work, it reduces the impact of this problem

**Comment by ecrist** [comment]
This should be fixed now after #312 was applied.

**Comment by luizluca** [comment]
More like a "workaround" than a real fix but this issue will not happen.
